tests/v2/validation/upgrade/workload.go

```diff
@@ -244,17 +244,33 @@ func checkPrefix(name string, prefix string) bool {
 	return strings.HasPrefix(name, prefix)
 }
 
-// validateDaemonset checks daemonset available number is equal to the number of workers in the cluster
+// validateDaemonset checks daemonset available number is equal to the number of workers in the downstream cluster
+// validateDaemonset checks daemonset available number is equal to the number of nodes in the local cluster
 func validateDaemonset(t *testing.T, client *rancher.Client, clusterID, namespaceName, daemonsetName string) {
 	t.Helper()
 
-	workerNodesCollection, err := client.Management.Node.List(&types.ListOpts{
-		Filters: map[string]interface{}{
-			"clusterId": clusterID,
-			"worker":    true,
-		},
-	})
-	require.NoError(t, err)
+	var workerNodesCollection *management.NodeCollection
+	var nodesCollection *management.NodeCollection
+	var err error
```

Comment:
Is there a specific reason for this? Why not just have it defined later on during the `workerNodesCollection`?

Reply:
Are you referring to the variable declarations? 🤔 

The reasoning for this is to allow those variables to be available outside the scope of the "if" statements, which is needed for the assert checks on lines 287 and 291.

Reply:
Ah, missed the asserts reusing. That answers my question, thanks!